lib/crawler.js

```diff
@@ -224,28 +383,44 @@ var Crawler = function(initialURL) {
         /^font\//i
     ];
 
-    // Whether to allow 'resources' greater than the max depth to be downloaded
-    crawler.fetchWhitelistedMimeTypesBelowMaxDepth = false;
-
-    // Ignore invalid SSL certificates
-    crawler.ignoreInvalidSSL = false;
-
-    // The HTTP / HTTPS agent used to crawl
-    crawler.httpAgent       = http.globalAgent;
-    crawler.httpsAgent      = https.globalAgent;
+    /**
+     * Controls whether resources with a whitelisted mimetype should be fetched,
+     * even though they are below the configured {@link Crawler#maxDepth}
+     * @type {Boolean}
+     */
+    this.fetchWhitelistedMimeTypesBelowMaxDepth = false;
+
+    /**
+     * Controls whether to proceed anyway when the crawler encounters an invalid
+     * SSL certificate.
+     * @type {Boolean}
+     */
+    this.ignoreInvalidSSL = false;
+
+    /**
+     * Controls what HTTP agent to use. This is useful if you want to configure
+     * eg. a SOCKS client.
+     * @type {HTTPAgent}
+     */
+    this.httpAgent = http.globalAgent;
+
+    /**
+     * Controls what HTTPS agent to use. This is useful if you want to configure
+     * eg. a SOCKS client.
+     * @type {HTTPAgent}
+     */
+    this.httpsAgent = https.globalAgent;
 
     // STATE (AND OTHER) VARIABLES NOT TO STUFF WITH
     var hiddenProps = {
-        _robotsTxts: [],
+        _fetchConditions: [],
         _isFirstRequest: true,
+        _openListeners: 0,
         _openRequests: [],
-        _fetchConditions: [],
-        _openListeners: 0
+        _robotsTxts: [],
+        _touchedHosts: []
```

Comment:
I made `touchedHosts` a "private" member, since it really is meant only for internal use by the crawler